ores/scoring_systems/celery_queue.py

```diff
@@ -0,0 +1,287 @@
+                        self.application.backend.mark_as_failure(
+                            task_id, RuntimeError("Never started"))
+                    continue
+                root_cache = {str(k): v for k, v in root_caches[rev_id].items()}
+                result = self._process_score_map.delay(
+                    context_name, missing_models, rev_id, root_cache,
+                    injection_cache=injection_cache,
+                    include_features=include_features)
+
+                for model_name in missing_models:
+                    task_id = context.format_id_string(
+                        model_name, rev_id, injection_cache=injection_cache)
+                    score_result = self._lookup_score_in_map.apply_async(
+                        args=(result.id, model_name), task_id=task_id,
+                        expires=self.timeout)
+                    if rev_id in results:
+                        results[rev_id][model_name] = score_result
+                    else:
+                        results[rev_id] = {model_name: score_result}
+
+        # Read results
+        rev_scores = {}
+        errors = {}
+        combined_results = chain(inprogress_results.items(), results.items())
+        for rev_id, model_results in combined_results:
+            if rev_id not in rev_scores:
+                rev_scores[rev_id] = {}
+            for model_name, score_result in model_results.items():
+                try:
+                    rev_scores[rev_id][model_name] = \
+                        score_result.get(timeout=self.timeout)
+                except celery.exceptions.TimeoutError:
+                    errors[rev_id] = TimeoutError(
+                        "Timed out after {0} seconds.".format(self.timeout))
+                    self.application.backend.mark_as_failure(
+                        score_result.id, TimeoutError("Timed out"))
+                except Exception as error:
+                    errors[rev_id] = error
+
+        return rev_scores, errors
+
+    def _lookup_inprogress_results(self, context_name, model_names, rev_ids,
+                                   injection_caches=None, rev_scores=None):
+        context = self[context_name]
+
+        inprogress_results = {}
+        for rev_id in rev_ids:
+            injection_cache = injection_caches.get(rev_id) \
+                              if injection_caches is not None else None
+
+            for model_name in model_names:
+                if rev_id in rev_scores and \
+                   model_name in rev_scores[rev_id]:
+                    continue
+
+                task_id = context.format_id_string(
+                    model_name, rev_id, injection_cache=injection_cache)
+                score_result = \
+                    self._lookup_score_in_map.AsyncResult(task_id)
+                if score_result.state in ("REQUESTED", "SENT", "STARTED",
+                                          "SUCCESS"):
```

Comment:
These states should be defined as constants (for `REQUESTED` and `SENT`) and referenced from `celery.states` for the others